File: 백준/Silver/28345. Image Filter/Image Filter.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int t, r, c, A, B, C, minn, idx, *temp, **arr, **rlt;
    temp = (int*)malloc(sizeof(int) * 5);
    scanf("%d", &t);
    for (int i = 0; i < t; i++) {
        scanf("%d %d", &r, &c);
        arr = (int**)malloc(sizeof(int*) * r);
        rlt = (int**)malloc(sizeof(int*) * r);
        for (int j = 0; j < r; j++) {
            arr[j] = (int*)malloc(sizeof(int) * c);
            for (int k = 0; k < c; k++) {
                scanf("%d", &arr[j][k]);
            }
        }
        for (int j = 0; j < r; j++) {
            rlt[j] = (int*)malloc(sizeof(int) * (c * 2));
            idx = 0;
            for (int k = 0; k < c; k++) {
                A = 0, B = 0, C = 0, minn = pow(2, 16);
                if (j == 0 && k > 0) {
                    A = arr[j][k - 1];
                } else if (j > 0 && k == 0) {
                    B = arr[j - 1][k];
                } else if (j > 0 && k > 0) {
                    A = arr[j][k - 1];
                    B = arr[j - 1][k];
                    C = arr[j - 1][k - 1];
                }
                temp[0] = arr[j][k] - 0;
                temp[1] = arr[j][k] - A;
                temp[2] = arr[j][k] - B;
                temp[3] = arr[j][k] - (A + B) / 2;
                temp[4] = arr[j][k] - (A + B - C);
                for (int l = 0; l < 5; l++) {
                    if (minn > abs(temp[l]))
                        minn = abs(temp[l]);
                }
                for (int l = 0; l < 5; l++) {
                    if (minn == abs(temp[l])) {
                        rlt[j][idx++] = l;
                        rlt[j][idx++] = temp[l];
                        break;
                    }
                }
            }
        }
        for (int j = 0; j < r; j++) {
            for (int k = 0; k < c * 2; k++) {
                printf("%d ", rlt[j][k]);
            }
            printf("\n");
        }
    }
    return 0;
}

File: 백준/Silver/28345. Image Filter/Image Filter.py
import sys
input=lambda:sys.stdin.readline().rstrip()
t=int(input())
for _ in range(t):
    rlt=[]
    r,c=map(int,input().split())
    arr=[list(map(int,input().split())) for _ in range(r)]
    for i in range(r):
        rlt.append([])
        for j in range(c):
            temp,minn,A,B,C=[],float("INF"),0,0,0
            if i==0 and j>0:
                A=arr[i][j-1]
            elif i>0 and j==0:
                B=arr[i-1][j]
            elif i>0 and j>0:
                A=arr[i][j-1]
                B=arr[i-1][j]
                C=arr[i-1][j-1]
            temp+=[arr[i][j]-0,arr[i][j]-A,arr[i][j]-B,arr[i][j]-(A+B)//2,arr[i][j]-(A+B-C)]
            for k in temp:
                if minn>abs(k):
                    minn=abs(k)
            for l,k in enumerate(temp):
                if minn==abs(k):
                    rlt[i].append(l)
                    rlt[i].append(k)
                    break
    for i in range(r):
        print(*rlt[i])